Apps/W1/APIV2/test/src/APIV2FixedAssetsE2E.Codeunit.al

```diff
@@ -0,0 +1,177 @@
+        TargetURL := LibraryGraphMgt.CreateTargetURL('', Page::"APIV2 - Fixed Assets", ServiceNameTxt);
+        LibraryGraphMgt.PostToWebService(TargetURL, GetFixedAssetJSON(FixedAsset), Response);
+
+        // [THEN] The Fixed Asset has been created in the database with all the details.
+        FixedAsset.Get(FixedAsset."No.");
+        VerifyProperties(Response, FixedAsset);
+    end;
+
+    [Test]
+    procedure TestDeleteFixedAsset()
+    var
+        FixedAsset: Record "Fixed Asset";
+        FixedAssetNo: Code[20];
+        TargetURL: Text;
+        Response: Text;
+    begin
+        // [SCENARIO] User can delete a Fixed Asset by making a DELETE request.
+        Initialize();
+
+        // [GIVEN] A Fixed Asset exists.
+        LibraryFixedAsset.CreateFixedAsset(FixedAsset);
+        FixedAssetNo := FixedAsset."No.";
+
+        // [WHEN] The user makes a DELETE request to the endpoint for the Fixed Asset.
+        TargetURL := LibraryGraphMgt.CreateTargetURL(FixedAsset.SystemId, Page::"APIV2 - Fixed Assets", ServiceNameTxt);
+        LibraryGraphMgt.DeleteFromWebService(TargetURL, '', Response);
+
+        // [THEN] The response is empty.
+        Assert.AreEqual('', Response, 'DELETE response should be empty.');
+
+        // [THEN] The Fixed Asset is no longer in the database.
+        FixedAsset.SetRange("No.", FixedAssetNo);
+        Assert.IsTrue(FixedAsset.IsEmpty(), 'Fixed Asset should be deleted.');
+    end;
+
+    [Test]
+    procedure TestModifyFixedAsset()
+    var
+        FixedAsset: Record "Fixed Asset";
+        TempFixedAsset: Record "Fixed Asset" temporary;
+        RequestBody: Text;
+        Response: Text;
+        TargetURL: Text;
+    begin
+        // [SCENARIO] User can modify a Fixed Asset through a PATCH request.
+        Initialize();
+
+        // [GIVEN] A Fixed Asset exists.
+        LibraryFixedAsset.CreateFixedAsset(FixedAsset);
+        TempFixedAsset.TransferFields(FixedAsset);
+        TempFixedAsset.Description := LibraryUtility.GenerateGUID();
+        RequestBody := GetFixedAssetJSON(TempFixedAsset);
+
+        // [WHEN] The user makes a patch request to the service.
+        TargetURL := LibraryGraphMgt.CreateTargetURL(FixedAsset.SystemId, Page::"APIV2 - Fixed Assets", ServiceNameTxt);
+        LibraryGraphMgt.PatchToWebService(TargetURL, RequestBody, Response);
+
+        // [THEN] The response text contains the new values.
+        VerifyProperties(Response, TempFixedAsset);
+
+        // [THEN] The record in the database contains the new values.
+        FixedAsset.Get(FixedAsset."No.");
+        VerifyProperties(Response, FixedAsset);
+    end;
+
+    local procedure GetFixedAssetJSON(var FixedAsset: Record "Fixed Asset"): Text
+    var
+        FixedAssetJson: Text;
+    begin
+        if FixedAsset."No." = '' then
+            FixedAsset."No." := NextFixedAssetNo();
+        if FixedAsset.Description = '' then
+            FixedAsset.Description := LibraryUtility.GenerateGUID();
```

Comment:
Same as for FA locations above - it doesn't look like these conditions are ever satisfied, code would be cleaner without them.

Reply:
It is written the same way as standard tests so for not confusing if the one test is written different way as other I think it should stay as is.

![image](https://github.com/microsoft/ALAppExtensions/assets/106300783/ebb32344-6912-47f4-b4fd-5559d34c7ab0)